### ISSUE TITLE
household() will return IDs that don't meet the criteria for inclusion

### ISSUE THREAD

**Issue by tarakc02** [tarakc02/discoveryengine#16]
Right now household returns the primary household ID, regardless of whether that ID was part of the list before. Traditionally we leave IDs as is, unless both spouses are in the list in which case we keep just the  primary ID.

**Comment by tarakc02** [comment]
I decided to take another shot at this, and in the process realized I'm not really sure what `household` is supposed to do. In particular, should it (a) return the ID that uniquely and consistently defines the household, or (b) return the original list, having removed pseudo-duplicates (by which I mean, cases where two IDs from the same household appear on the original list). (b) is what the prospect pool spreadsheet does when it households. 

The fact that there is no separate set of household IDs (rather, exactly one ID is given the status of "primary" which means it can stand for the household) makes this distinction invisible most of the time. 

The case for (a) comes from a recent issue that LC had. She wanted to pull two separate lists, one of school of information graduates, and the other of donors to SIM who are not degreeholders. When she first ran this, she noticed that the latter list was including spouses of degreeholders. I suggested she household both lists so that when she did `donor %but_not% degreeholder` it would be excluding entire households. That presupposes interpretation (a). 

I think that example demonstrates that if you need to do boolean calculus with the lists (which is the whole purpose of the discoveryengine) then you have to use interpretation (a). But for output purposes, interpretation (b) is correct (that's why we do it that way in the PP spreadsheet report). So I think it is correct to keep the `household` converter as it is. The interesting next question is whether the household option in `display` should do (b). The case against doing that is the additional complexity it adds. Like, is it reasonable that:

```
display(household(definition), household = FALSE)
```

should have different results from:

```
display(definition, household = TRUE)
```

???

**Comment by tarakc02** [comment]
Just a comment to point out that this issue is only really an issue when it comes to giving totals, so that a fix to #53 will mostly make this one moot. So I'm closing this and will focus on #53
